app/validation/validator.py

```diff
@@ -122,80 +122,79 @@ def validate_answer(self, block, answer):
-
-        return range_errors
-
-    def validate_range_value(self, value, answer_id, answer_decimals):
-        error_message = 'Decimal Places used in {} should be less than or equal to {}, currently {}' \
-            .format(answer_id, MAX_DECIMAL_PLACES, answer_decimals)
-
-        if answer_decimals > MAX_DECIMAL_PLACES:
-            return [self._error_message(error_message)]
-
-        error_message = 'Value {} used in {} should be between system limits {} to {}'\
-            .format(value, answer_id, MIN_NUMBER, MAX_NUMBER)
-
-        if MIN_NUMBER <= value <= MAX_NUMBER:
-            return []
-
-        return [self._error_message(error_message)]
+    def _get_answer_range_values(self, answer, answer_ranges):
+
+        return {
+            'min': self._get_answer_minimum_or_maximum(answer['id'],
+                                                       answer_ranges,
+                                                       answer.get('min_value', {'value': 0}),
+                                                       answer.get('decimal_places', 0),
+                                                       'minimum'),
+            'max': self._get_answer_minimum_or_maximum(answer['id'],
+                                                       answer_ranges,
+                                                       answer.get('max_value', {'value': MAX_NUMBER}),
+                                                       answer.get('decimal_places', 0),
+                                                       'maximum'),
+            'decimal_places': answer.get('decimal_places', 0),
+        }
+
+    def _get_answer_minimum_or_maximum(self, answer_id, answer_ranges, attributes, decimal_places, min_or_max):
+        values = []
+        if 'value' in attributes:
+            if not MIN_NUMBER <= attributes.get('value') <= MAX_NUMBER:
+                error_message = 'Value {} used in {} should be between system limits {} to {}' \
+                    .format(attributes['value'], answer_id, MIN_NUMBER, MAX_NUMBER)
+                raise ValidationError(error_message)
+            values = [attributes.get('value')]
+
+        elif 'answer_id' in attributes:
+            if decimal_places < answer_ranges[attributes['answer_id']]['decimal_places']:
+                error_message = '{} has {} decimal places and is being passed to max/min value for {} of {} ' \
+                                'decimal_places. The referred answer cannot have more decimal places'\
+                    .format(attributes['answer_id'], answer_ranges[attributes['answer_id']]['decimal_places'],
+                            answer_id, decimal_places)
+                raise ValidationError(error_message)
+            values = self._get_range_for_referred_answer(answer_ranges, attributes['answer_id'])
+
+        for idx, value in enumerate(values):
+            if isinstance(value, list):
+                values[idx] = value[0]
```

Comment:
I don't really understand what these 3 lines are doing?

Reply:
So when there is an answer_id the method goes and gets that answers [[min], [max]] and they form the new minimum or maximum as possibilities, but if its just a value its returned as [min] or [max].
So these lines de-list and then put into a new list, which means that math operations can occur when they are exclusive or not

Reply:
but if the value is a list it only uses the 1st value? therefore why create a list in the first place if only the 1st value will be used.